### ISSUE TITLE
Migrate Linux-based CI jobs to GitHub Actions

### ISSUE THREAD

**Issue by mojavelinux** [asciidoctor/asciidoctor#3837]
Travis CI has rescinded its commitment to open source projects by enforcing quotas.

> We will be offering an allotment of OSS minutes that will be reviewed and allocated on a case by case basis.

This is unacceptable to loyal open source users like the Asciidoctor project, which has advocated strongly for their platform over the years. The performance of our jobs has also degraded measurably in recent months. The last release took over 24 hours to complete due to the fact that jobs took several hours to start. I can only conclude this is retaliation for not migrating from travis-ci.org to travis-ci.com, a transition which has proved unreasonably complex and confusing.

In short, I'm unhappy with Travis CI and the direction it's headed, so I intend to migrate this project away from it. The logical choice is GitHub Actions because several projects in the ecosystem have already been having success with it. We can research and explore other options once this migration is complete.

Since GitHub Actions supports Windows, I'd also like to migrate the Windows jobs from AppVeyor to GitHub Actions as well to bring all the jobs together. This will speed up the turnaround on PRs and releases. We will handle that migration in a separate issue.

**Comment by mojavelinux** [comment]
The Linux-based builds have been migrated. These also need to go to the 1.5.x branch as well.

**Comment by mojavelinux** [comment]
> These also need to go to the 1.5.x branch as well.

Actually, that branch is now defunct. We should just remove the CI build.
